packages/create-cloudflare/e2e-tests/pages.test.ts

```diff
@@ -1,63 +1,67 @@
 import { existsSync, mkdtempSync, realpathSync, rmSync } from "fs";
+import crypto from "node:crypto";
 import { tmpdir } from "os";
 import { join } from "path";
 import { FrameworkMap } from "frameworks/index";
 import { readJSON } from "helpers/files";
+import { fetch } from "undici";
 import { describe, expect, test, afterEach, beforeEach } from "vitest";
 import { keys, runC3 } from "./helpers";
 import type { RunnerConfig } from "./helpers";
+import spawn from "cross-spawn";
 
 /*
 Areas for future improvement:
-- Make these actually e2e by verifying that deployment works
 - Add support for frameworks with global installs (like docusaurus, gatsby, etc)
 */
 
+type FrameworkTestConfig = RunnerConfig & {
+	expectResponseToContain: string;
+};
+
 describe("E2E: Web frameworks", () => {
 	const tmpDirPath = realpathSync(mkdtempSync(join(tmpdir(), "c3-tests")));
-	const projectPath = join(tmpDirPath, "pages-tests");
+	let projectPath: string;
+	let projectName: string;
 
 	beforeEach(() => {
+		projectName = `c3-e2e-${crypto.randomBytes(4).toString("hex")}`;
+		projectPath = join(tmpDirPath, projectName);
 		rmSync(projectPath, { recursive: true, force: true });
 	});
 
 	afterEach(() => {
 		if (existsSync(projectPath)) {
 			rmSync(projectPath, { recursive: true });
 		}
+
+		spawn("npx", ["wrangler", "pages", "project", "delete", "-y", projectName]);
```

Comment:
Yes, and yes. I think we'd need to make a separate script pipeline for the cleanup so it could be run on its own and perhaps on a schedule. It's pretty straightforward though now that we have a `wrangler pages project delete` command